### PR TITLE
Devtooling-356: refactored group roles resource

### PR DESCRIPTION
**Pull request by BrianMoyles**

### CODE REVIEW

File: genesyscloud/group_roles/genesyscloud_group_roles_init_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	providerResources = make(map[string]*schema.Resource)
+
+	regInstance := &registerTestInstance{}
+
+	regInstance.registerTestResources()
+}
+
+// TestMain is a "setup" function called by the testing framework when run the test
+func TestMain(m *testing.M) {
+	// Run setup function before starting the test suite for outbound_callabletimeset package
```
Comment (dginty4) [inline genesyscloud/group_roles/genesyscloud_group_roles_init_test.go:44]: Incorrect comment

File: genesyscloud/group_roles/resource_genesyscloud_group_roles_schema.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+const resourceName = "genesyscloud_group_roles"
+
+// SetRegistrar registers all of the resources and exporters in the package
+func SetRegistrar(l registrar.Registrar) {
+	l.RegisterResource(resourceName, ResourceGroupRoles())
+	l.RegisterExporter(resourceName, GroupRolesExporter())
+}
+
+// ResourceOutboundCallabletimeset registers the genesyscloud_group_roles resource with Terraform
```
Comment (dginty4) [inline genesyscloud/group_roles/resource_genesyscloud_group_roles_schema.go:18]: Incorrect comment

File: genesyscloud/group_roles/resource_genesyscloud_group_roles.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+func createGroupRoles(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	groupID := d.Get("group_id").(string)
+	d.SetId(groupID)
+	return updateGroupRoles(ctx, d, meta)
+}
+
+// readGroupRoles is used by the group_roles resource to read Group Roles from the genesys cloud
+func readGroupRoles(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*genesyscloud.ProviderMeta).ClientConfig
+	authAPI := platformclientv2.NewAuthorizationApiWithConfig(sdkConfig)
```
Comment (carnellj-genesys) [inline genesyscloud/group_roles/resource_genesyscloud_group_roles.go:31]: We need to add a Proxy class.  We should not be calling the API directly from the code.

File: genesyscloud/group_roles/resource_genesyscloud_group_roles.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		_ = d.Set("roles", roles)
+
+		log.Printf("Read roles for group %s", d.Id())
+		return cc.CheckState()
+	})
+}
+
+func updateGroupRoles(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*genesyscloud.ProviderMeta).ClientConfig
+	authAPI := platformclientv2.NewAuthorizationApiWithConfig(sdkConfig)
```
Comment (carnellj-genesys) [inline genesyscloud/group_roles/resource_genesyscloud_group_roles.go:56]: Proxy needed

**Review by carnellj-genesys — CHANGES_REQUESTED** [review]

File: genesyscloud/user_group_role_common.go [hunk 5]
```diff
@@ -53,7 +53,7 @@ func getAssignedGrants(subjectID string, authAPI *platformclientv2.Authorization
 	return grants, resp, nil
 }
 
-func readSubjectRoles(d *schema.ResourceData, authAPI *platformclientv2.AuthorizationApi) (*schema.Set, *platformclientv2.APIResponse, diag.Diagnostics) {
+func ReadSubjectRoles(d *schema.ResourceData, authAPI *platformclientv2.AuthorizationApi) (*schema.Set, *platformclientv2.APIResponse, diag.Diagnostics) {
```
Comment (HemanthDogiparthi12) [inline genesyscloud/user_group_role_common.go:56]: move this file as well to its own package. most likely this wil be used by users and groups. and add a proxy for this.

**Pull request merged**